src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.SymbolSet.cs

```diff
@@ -0,0 +1,270 @@
+                ImmutableHashSet<Project> projects, CancellationToken cancellationToken)
+            {
+                Contract.ThrowIfFalse(projects.Count == 1, "Only a single project should be passed in");
+
+                // Don't bother on symbols that aren't even involved in inheritance computations.
+                if (!InvolvesInheritance(symbol))
+                    return;
+
+                var solution = engine._solution;
+                if (symbol.IsImplementableMember())
+                {
+                    var implementations = await SymbolFinder.FindMemberImplementationsArrayAsync(
+                        symbol, solution, projects, cancellationToken).ConfigureAwait(false);
+
+                    await AddCascadedAndLinkedSymbolsToAsync(engine, implementations, seenSymbols, workQueue, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    var overrrides = await SymbolFinder.FindOverridesArrayAsync(
+                        symbol, solution, projects, cancellationToken).ConfigureAwait(false);
+
+                    await AddCascadedAndLinkedSymbolsToAsync(engine, overrrides, seenSymbols, workQueue, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            /// <summary>
+            /// Finds all the symbols 'up' the inheritance hierarchy of <paramref name="symbol"/> in the solution.  The
+            /// symbols found are added to <paramref name="seenSymbols"/>.  If <paramref name="seenSymbols"/> did not contain that symbol,
+            /// then it is also added to <paramref name="workQueue"/> to allow fixed point algorithms to continue.
+            /// </summary>
+            protected static async Task AddUpSymbolsAsync(
+                FindReferencesSearchEngine engine, ISymbol symbol,
+                HashSet<ISymbol> seenSymbols, Stack<ISymbol> workQueue,
+                ImmutableHashSet<Project> projects, CancellationToken cancellationToken)
+            {
+                if (!InvolvesInheritance(symbol))
+                    return;
+
+                var solution = engine._solution;
+                var originatingProject = solution.GetOriginatingProject(symbol);
+                if (originatingProject != null)
+                {
+                    // We have a normal method.  Find any interface methods up the inheritance hierarchy that it implicitly
+                    // or explicitly implements and cascade to those.
+                    foreach (var match in await SymbolFinder.FindImplementedInterfaceMembersArrayAsync(symbol, solution, projects, cancellationToken).ConfigureAwait(false))
+                        await AddCascadedAndLinkedSymbolsToAsync(engine, match, seenSymbols, workQueue, cancellationToken).ConfigureAwait(false);
+                }
+
+                // If we're overriding a member, then add it to the up-set
+                if (symbol.GetOverriddenMember() is { } overriddenMember)
```

Comment:
```suggestion
                if (symbol.GetOverriddenMember() is ISymbol overriddenMember)
```

Reply:
While the braces are nifty, I think they're harder to interpret than just using a type outright.

Reply:
done.